src/artm/core/batch_manager.h

```diff
@@ -46,6 +46,8 @@ class BatchManager : boost::noncopyable, public Notifiable {
   // until it is marked as processed by Done().
   BatchManagerTask Next();
 
+  void AddAndNext(const BatchManagerTask& task);
```

Comment:
Overall I don't like method names in BatchManager -- Add, Next, and now AddAndNext makes very little sense to anyone (including me).
- Add() adds the a passive task to the queue
- Next() finds some passive task and moves it to the active queue (for each model)
- AddAndNext() puts the task directly to the active queue (for each model)
- Done() marks an active task as "done" (for a given model)
If you can find a better names for this operations, please rename. If not, just copy my test above as a comment into the code :)